### PR TITLE
Added a filter for the trans tag

### LINKED ISSUE #10126: Document the built in way to use trans/translate in a Django template filter
**UPDATE: This is now mostly a documentation piece, the pattern (see comments) is good and maybe a PR with documentation updates AND a few changes to existing templates is all that is needed.**

### Is your proposal related to a problem?

We often find ourselves having to build translated values with `{% trans ... as foo %}`, not a fragment but for use in some other variable passed into a template.

As we adopt more shared template tags, this is becoming a bit of code smell and it would be great to be able to simplify this approach.

### Describe the solution you'd like

It would be great if we could do something like the following:

```html
{% status url=page_url title="Visit the live page"|trans hidden_label="Current page status"|trans classname="primary" %}
```

This would then be picked up by the translation system as if we had used

```html
{% trans "Current page status" as hidden_label %}
{% trans "Visit the live page" as status_label %}
{% status url=page_url title=status_label hidden_label=hidden_label classname="primary" %}
```

### Describe alternatives you've considered

Leave as is, keep declaring translate blocks /content in the vanilla Django way.

### Additional context

* _Originally posted by @gasman in https://github.com/wagtail/wagtail/issues/9721#issuecomment-1351503209_

### PR DESCRIPTION
**Pull request by AceHunterr**
Fixes #10126 

Added a filter for {%trans%} so as to simplify its use in the tags.

Although I've some issues regarding it:
1.I was not sure where to put the custom filter created so I put that in the [wagtail\wagtail\admin\templatetags](https://github.com/wagtail/wagtail/tree/main/wagtail/admin/templatetags). Please correct me if I've not put this in the right place.

2. I didn't know how to test the filter created so I don't know exactly how is it supposed to work.

P.S.: The pre-commit hook is showing errors for windows still... so the `.pre-commit-config.yaml` file can show some merge conflicts.... I've raised such issue in #10172

### CODE REVIEW

**Comment by lb-** [comment]
@AceHunterr please the fix for https://github.com/wagtail/wagtail/issues/10172 to a new PR. It will be easier to review in isolation. 

As for the new template tag. I think this file may be more suitable. 

https://github.com/wagtail/wagtail/blob/main/wagtail/admin/templatetags/wagtailadmin_tags.py

File: wagtail/admin/templatetags/trans_filter.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from django import template
+from django.utils.translation import gettext as _
+
+register = template.Library()
+
+
+@register.simple_tag
+def trans(msg_var):
+    result_var = msg_var
+    return _(result_var)
```
Comment (lb-) [inline wagtail/admin/templatetags/trans_filter.py:10]: Hey. This might work for known translations but I'm not sure this will work for new translations. 


See how make translations works. https://docs.djangoproject.com/en/4.1/ref/django-admin/#django-admin-makemessages

See also https://docs.djangoproject.com/en/4.1/topics/i18n/translation/

Comment (lb-) [reply wagtail/admin/templatetags/trans_filter.py:10]: I might suggest starting with a few unit tests that also check whether the make translations script will parse them.

**Comment by AceHunterr** [comment]
I'm sorry but I didn't understood what you meant by "Review in Isolation" ... Like what I have to do?

**Comment by lb-** [comment]
> I'm sorry but I didn't understood what you meant by "Review in Isolation" ... Like what I have to do?

This PR contains fixes for two completely different issues, it's best if one PR fixes one issue at a time. I recommend you create a new branch for the pre-commit fix and then push that fix up, then create a standalone PR for that.

Hope that makes sense, it seems tedious at first but does make it much easier for the core team to review and merge a PR if it's as small as possible.

**Comment by AceHunterr** [comment]
I've put the trans filter created in the correct file and changed the name of the function to 'translate'

**Comment by lb-** [comment]
@AceHunterr have a look at this comment. https://github.com/wagtail/wagtail/issues/10126#issuecomment-1466937171

I'm pretty sure now we won't need a custom template filter but rather just some guidance in our translators docs and one or two updates to existing templates as a reference.

**Comment by AceHunterr** [comment]
> @AceHunterr have a look at this comment. [#10126 (comment)](https://github.com/wagtail/wagtail/issues/10126#issuecomment-1466937171)
> 
> I'm pretty sure now we won't need a custom template filter but rather just some guidance in our translators docs and one or two updates to existing templates as a reference.

Okay yeah I did saw that comment and the code.

**Comment by AceHunterr** [comment]
Hey @lb- I changed the [wagtail/admin/templates/wagtailadmin/shared/page_status_tag.html](https://github.com/wagtail/wagtail/blob/main/wagtail/admin/templates/wagtailadmin/shared/page_status_tag.html) the way you described.
I also changed the description in the `docs/contributing/translations.md` ... Although I'm not very sure if I had made the changes in the most decent way but I couldn't get any other way of explaining the alternative for the {%translate%} tag in the translations.md file... So if any other changes lemme know

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,33 +1,33 @@
 default_language_version:
-  node: system
```
Comment (lb-) [inline .pre-commit-config.yaml:2]: A reminder to please revert the changes to this file so this PR.

File: docs/contributing/translations.md [hunk 3]
```diff
@@ -66,6 +66,19 @@ _("Page {page_title} with status {status}".format(page_title=page.title, status=
 
 # Do not use f-string
 _(f"Page {page.title} with status {page.status_string}")
+
+# Do not use {%translate%} or {%blocktranslate%}
```
Comment (lb-) [inline docs/contributing/translations.md:70]: This is probably not as clear as it could be, and this should probably be in its own sub-heading section.

Additionally, you have the code example in a `python` block, but the template tags are used in `html` blocks.

````md

### Passing translated strings into template tag variables

Instead of setting a variable that is the translated string and passing the variable into the template tag as below;

```html
{% comment %} Avoid doing this {% endcomment %}
{% load i18n wagtailadmin_tags %}
{% pageurl page as page_url %}
{% trans 'Visit the live page' as status_title %}
{% status page.status_string url=page_url title=status_title classname="primary" %}
```

it is better to leverage Django's documented approach to passing [translated strings to template tags and filters](https://docs.djangoproject.com/en/4.2/topics/i18n/translation/#string-literals-passed-to-tags-and-filters)

```html
{% comment %} Instead do this {% endcomment %}
{% load i18n wagtailadmin_tags %}
{% pageurl page as page_url %}
{% status page.status_string url=page_url title=_("Visit the live page") classname="primary" %}
```

This not only makes the template code easier to read, it also avoids risks of polluting template variables for other code.

````

Note: The django docs link will not be suitable, you will need to see how we link to Django docs in other parts of code and follow that.

File: wagtail/admin/templatetags/wagtailadmin_tags.py [hunk 4]
```diff
@@ -131,6 +131,12 @@ def fieldtype(bound_field):
             return ""
 
 
+@register.filter
```
Comment (lb-) [inline wagtail/admin/templatetags/wagtailadmin_tags.py:134]: This change is no longer needed.

File: wagtail/admin/templates/wagtailadmin/shared/page_status_tag.html [hunk 5]
```diff
@@ -1,14 +1,9 @@
 {% load i18n wagtailadmin_tags %}
 {% if page.live %}
     {% with page_url=page.url %}
-        {% trans "Current page status:" as status_hidden_label %}
```
Comment (lb-) [inline wagtail/admin/templates/wagtailadmin/shared/page_status_tag.html:4]: This file will be updated in #10157 so please maybe find another one or two files to update with this convention.

**Review by lb- — CHANGES_REQUESTED** [review]
@AceHunterr  some additional feedback, let us know if you are ok to keep pursuing this PR.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (AceHunterr) [reply .pre-commit-config.yaml:2]: My git is not commiting my code if I put the pre-commit-config file back to the default one

**Comment by AceHunterr** [comment]
I've actually implemented the changes but I'm having troubles pushing the code I'll update as soon as that error is fixed.

**Comment by lb-** [comment]
@AceHunterr I think it may be best to close this for now, as we no longer need these actual code changes.

If you are still keen on doing just some documentation for how to use this pattern, please raise another PR.

**Comment by AceHunterr** [comment]
I'm sincerely apologetic I couldn't continue working on this due to my prior engagements and couldn't contribute much in the past month ;(

**Comment by lb-** [comment]
@AceHunterr it's not a problem at all, thanks for giving this a solid go.
